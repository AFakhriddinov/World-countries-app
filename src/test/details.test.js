import React from 'react';
import { useSelector } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HomePage from '../components/homePage';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('<HomePage />', () => {
  const mockedArray = [
    { name: 'Test Country 1', flag: 'https://test-country-1-flag.png' },
    { name: 'Test Country 2', flag: 'https://test-country-2-flag.png' },
    { name: 'Test Country 3', flag: 'https://test-country-3-flag.png' },
  ];

  beforeEach(() => {
    useSelector.mockReturnValue({ StateInfo: mockedArray });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should display correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should display all states after loading', () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    const images = screen.getAllByRole('link', {
      name: /^test country/i,
    });
    expect(images).toHaveLength(mockedArray.length);
  });

  it('Should display states depending on filtered search', () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    const search = screen.getByPlaceholderText('Search a country here');
    fireEvent.change(search, { target: { value: 'test country 2' } });

    const stateImages = screen.getAllByText(/test country 2/i);
    expect(stateImages).toHaveLength(1);
  });
});
