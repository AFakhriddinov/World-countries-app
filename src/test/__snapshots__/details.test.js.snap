// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> Should display correctly 1`] = `
Array [
  <div
    className="header"
  >
    <img
      alt="banner-img"
      className="contryImg "
      src="backround.e8e3d87f186997fd15d8.png"
    />
  </div>,
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="Search a country here"
    type="text"
    value=""
  />,
  <div
    className="status-bar"
  >
    <h2
      className="country-status"
    >
      Stats by Country
    </h2>
  </div>,
  <div
    className="div"
  >
    <a
      className="container"
      href="/countries/Test Country 1"
      onClick={[Function]}
    >
      <img
        alt="Test Country 1"
        className="flag"
        src="https://test-country-1-flag.png"
      />
      <div
        className="info"
      >
        <h2
          className="name"
        >
          Test Country 1
        </h2>
        <h2
          className="population"
        />
      </div>
    </a>
    <a
      className="container"
      href="/countries/Test Country 2"
      onClick={[Function]}
    >
      <img
        alt="Test Country 2"
        className="flag"
        src="https://test-country-2-flag.png"
      />
      <div
        className="info"
      >
        <h2
          className="name"
        >
          Test Country 2
        </h2>
        <h2
          className="population"
        />
      </div>
    </a>
    <a
      className="container"
      href="/countries/Test Country 3"
      onClick={[Function]}
    >
      <img
        alt="Test Country 3"
        className="flag"
        src="https://test-country-3-flag.png"
      />
      <div
        className="info"
      >
        <h2
          className="name"
        >
          Test Country 3
        </h2>
        <h2
          className="population"
        />
      </div>
    </a>
  </div>,
]
`;

exports[`<HomePage /> Should render correctly 1`] = `
Array [
  <div
    className="header"
  >
    <img
      alt="banner-img"
      className="contryImg "
      src="backround.e8e3d87f186997fd15d8.png"
    />
  </div>,
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="Search a country here"
    type="text"
    value=""
  />,
  <div
    className="status-bar"
  >
    <h2
      className="country-status"
    >
      Stats by Country
    </h2>
  </div>,
  <div
    className="div"
  >
    <a
      className="container"
      href="/countries/Test Country 1"
      onClick={[Function]}
    >
      <img
        alt="Test Country 1"
        className="flag"
        src="https://test-country-1-flag.png"
      />
      <div
        className="info"
      >
        <h2
          className="name"
        >
          Test Country 1
        </h2>
        <h2
          className="population"
        />
      </div>
    </a>
    <a
      className="container"
      href="/countries/Test Country 2"
      onClick={[Function]}
    >
      <img
        alt="Test Country 2"
        className="flag"
        src="https://test-country-2-flag.png"
      />
      <div
        className="info"
      >
        <h2
          className="name"
        >
          Test Country 2
        </h2>
        <h2
          className="population"
        />
      </div>
    </a>
    <a
      className="container"
      href="/countries/Test Country 3"
      onClick={[Function]}
    >
      <img
        alt="Test Country 3"
        className="flag"
        src="https://test-country-3-flag.png"
      />
      <div
        className="info"
      >
        <h2
          className="name"
        >
          Test Country 3
        </h2>
        <h2
          className="population"
        />
      </div>
    </a>
  </div>,
]
`;
